### PR TITLE
esd can extended frame format

### PR DESCRIPTION
**Pull request by panxuechao**

### CODE REVIEW

File: modules/drivers/canbus/can_client/esd/BUILD [hunk 1]
```diff
@@ -13,6 +13,7 @@ cc_library(
     deps = [
         "//modules/drivers/canbus/can_client",
         "//third_party/can_card_library/esd_can",
+        "//modules/drivers/canbus:sensor_gflags",
```
Comment (xiaoxq) [inline modules/drivers/canbus/can_client/esd/BUILD]: Make it sorted.

File: modules/drivers/canbus/sensor_gflags.cc [hunk 2]
```diff
@@ -32,3 +32,7 @@ DEFINE_double(sensor_freq, 100,
 
 // System gflags
 DEFINE_string(sensor_node_name, "", "Sensor node name.");
+
+// esd can extended frame gflags
+DEFINE_bool(esd_can_extended_frame, false,
+                "check esd can exdended frame enabled or not");
```
Comment (xiaoxq) [inline modules/drivers/canbus/sensor_gflags.cc]: Align with first argument.

File: modules/tools/gen_vehicle_protocol/extract_dbc_meta.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 import shlex
 import re
 
-MAX_CAN_ID = 2048
+MAX_CAN_ID = 4096000000 #2048
```
Comment (xiaoxq) [inline modules/tools/gen_vehicle_protocol/extract_dbc_meta.py]: Two spaces before comments.

File: modules/tools/gen_vehicle_protocol/gen_protocols.py [hunk 4]
```diff
@@ -66,7 +66,11 @@ def gen_report_cpp(car_type, p, output_dir):
         fmt_val["protocol_name_lower"] = p["name"]
         classname = p["name"].replace('_', '').capitalize()
         fmt_val["classname"] = classname
-        fmt_val["id_upper"] = p["id"].upper()
+        a = int(p["id"].upper(),16)
```
Comment (xiaoxq) [inline modules/tools/gen_vehicle_protocol/gen_protocols.py]: 1. Space after `,`
2. Avoid single-char variable names. Please also fix the `p`.

File: modules/tools/gen_vehicle_protocol/gen_protocols.py [hunk 5]
```diff
@@ -449,6 +457,15 @@ def gen_protocols(protocol_conf_file, protocol_dir):
...
 
+def gen_esd_can_extended(s):
+    """
+        id string:
+    """
+    a = int(s,16)
+    a = a & 0x1FFFFFFF
+    a = a | 0x20000000
+    s = hex(a).replace('0x','')
+    return s
```
Comment (xiaoxq) [inline modules/tools/gen_vehicle_protocol/gen_protocols.py]: ditto

File: modules/tools/gen_vehicle_protocol/template/control_protocol.cc.tpl [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /******************************************************************************
- * Copyright 2018 The Apollo Authors. All Rights Reserved.
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
```
Comment (xiaoxq) [inline modules/tools/gen_vehicle_protocol/template/control_protocol.cc.tpl]: Don't change year of existing files.

**Comment by Capri2014** [comment]
@panxuechao  Nice work to extend the can message format to support extended frame! One thing is that this change would be applicable to all CAN card that supports extended frame (esd, hermes etc) and all CAN connected devices (vehicle, radar, etc), so could you rename the function name and remove the "esd" keyword to avoid confusion?

**Comment by Capri2014** [comment]
my bad, this is only for esd canard.

**Pull request merged**